torch/lib/TH/THSize.h

```diff
@@ -0,0 +1,13 @@
+#ifndef TH_SIZE_INC
+#define TH_SIZE_INC
+
+#include "THGeneral.h"
+#include <stddef.h>
+
+// THTensor functions that would work on a THSize if we had such a class in C++,
+// i.e. THTensor functions that depend only on the shape of the tensor, not the type.
```

Comment:
I think it's better to be consistent. Right now, we use LongStorage for such purposes, and I'd rather keep it like that, with an option to change it as you propose in a later PR.